src/librustc_trait_selection/traits/fulfill.rs

```diff
@@ -520,6 +522,68 @@ impl<'a, 'b, 'tcx> ObligationProcessor for FulfillProcessor<'a, 'b, 'tcx> {
                     Err(err) => ProcessResult::Error(CodeSelectionError(ConstEvalFailure(err))),
                 }
             }
+
+            ty::Predicate::ConstEquate(c1, c2) => {
+                debug!("equating consts: c1={:?} c2={:?}", c1, c2);
+
+                let stalled_on = &mut pending_obligation.stalled_on;
+
+                let mut evaluate = |c: &'tcx Const<'tcx>| {
```

Comment:
can we extract a helper for this code and the code in select.rs?

Reply:
I already tried this previously and was unable to do this/my solution made ended up being worse.

Reply:
OK, let's leave it for now I guess.